### PR TITLE
MAT-2564 Connect Virus Service to Bonnie QDM

### PR DESCRIPTION
**Pull request by ghost**
MAT-2564 Connect Virus Service to Bonnie QDM

There are 2 false positive codacy changes. **Please force merge**

Pull requests into Bonnie require the following. Submitter and reviewer should :white_check_mark: when done. For items that are not-applicable, note it's not-applicable ("N/A") and :white_check_mark:.

**Submitter:**
- [x] This pull request describes why these changes were made.
- [x] This PR is into the correct branch.
- [x] JIRA ticket for this PR:
- [ ] JIRA ticket links to this PR
- [x] Code diff has been done and been reviewed (it **does not** contain: additional white space, not applicable code changes, debug statements, etc.)
- [ ] If UI changes have been made, google WAVE plug-in has been executed to ensure no 508 issues were introduced.
- [ ] Tests are included and test edge cases
- [ ] Tests have been run locally and pass (remember to update Gemfile when applicable)
- [ ] Test fixtures updated and documented as necessary ( see [internal wiki](https://gitlab.mitre.org/bonnie/internal-documentation/wikis/testing#test-fixtures) )
- [ ] Code coverage has not gone down and all code touched or added is covered. 
     * In rare situations, this may not be possible or applicable to a PR. In those situations:
         1. Note why this could not be done or is not applicable here: 
         2. Add TODOs in the code noting that it requires a test
         3. Add a JIRA task to add the test and link it here: 
- [ ] Automated regression test(s) pass

If JIRA tests were used to supplement or replace automated tests:
- [ ] JIRA test links:
- [ ] Justification for using JIRA tests:
- [ ] JIRA tests have been added to sprint

**Reviewer 1:**

Name:
- [ ] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [ ] The tests appropriately test the new code, including edge cases
- [ ] You have tried to break the code

If JIRA tests were used to supplement or replace automated tests:
- [ ] JIRA tests have been run and pass
- [ ] You agree with the justification for use of JIRA tests or have provided input on why you disagree

**Reviewer 2:**

Name:
- [ ] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [ ] The tests appropriately test the new code, including edge cases
- [ ] You have tried to break the code

If JIRA tests were used to supplement or replace automated tests:
- [ ] JIRA tests have been run and pass
- [ ] You agree with the justification for use of JIRA tests or have provided input on why you disagree

### CODE REVIEW

File: app/helpers/virus_scan_helper.rb [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+      super(front_end_version: front_end_version, back_end_version: back_end_version, operator_error: true)
+    end
+  end
+
+  class MeasurePackageVirusFoundError < MeasureHelper::SharedError
+    def initialize
+      front_end_version = {
+        title: "Error Loading Measure",
+        summary: "The uploaded file is not a valid Measure Authoring Tool (MAT) export of a FHIR Based Measure.",
+        body: "Error: V100. Please re-package and re-export your FHIR based measure from the MAT and try again."
```
Comment (adongare) [inline app/helpers/virus_scan_helper.rb]: I think this needs to be specific to QDM measure

Comment (ghost) [reply app/helpers/virus_scan_helper.rb]: wow. good catch

File: app/helpers/virus_scan_helper.rb [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+      super(front_end_version: front_end_version, back_end_version: back_end_version, operator_error: true)
+    end
+  end
+
+  class PatientImportVirusFoundError < MeasureHelper::SharedError
+    def initialize
+      front_end_version = {
+        title: "Error Importing Patients",
+        summary: "The uploaded file is not a valid Bonnie patient export.",
+        body: "Error: V100. Please re-export patients from QDM bonnie and re-import in FHIR Bonnie."
```
Comment (adongare) [inline app/helpers/virus_scan_helper.rb]: Don't think this part is required here

Comment (ghost) [reply app/helpers/virus_scan_helper.rb]: Agree. I should have checked unused code

**Review by adongare — DISMISSED** [review]

**Review by adongare — APPROVED** [review]

**Review by AndrewBird81 — APPROVED** [review]
